### DISCUSSION TITLE
Localization

### DISCUSSION THREAD

**Discussion by johnmeshulam** [FIRSTIsrael/lems#729] [Ideas]
# LEMS Localization

This has been discussed as part of the growing initiative to bring LEMS outside of Israel. Localization includes all aspects of supporting events in a different location.

## What needs to be done?

Let's split up the tasks into categories:

### Language support

LEMS needs to be able to display its UI in any language. We are considering using [globalize](https://github.com/globalizejs/globalize) for this but other libraries are a welcome suggestion.

- [ ] Pick and install locale management library
- [ ] Move every single text in Hebrew to an he_il file and use references to that file in the current code
- [ ] Add other locale files
- [ ] Implement a switcher to easily switch between locales

### Location support

- [ ] Allow users to select the timezone that LEMS will generate schedules in
- [ ] Support for changing the date of events after creation

### Event format support

- [ ] Allow support for multi-day event schedules, as well as additional customization options that still need to be discussed

**Comment by TheCommandCat** [comment]
Hey @johnmeshulam , Just saw your branch on localization and your approach to scrape it.

I'm quite familiar with the "lingui" library and "crowdin platform" from other projects and gave it a quick try for localization here. I was able to get a basic setup working that might serve as a starting point.

I know the decision on the library is yours and the technical team, I wanted to share this experiment in case it offers any insights or a potential starting point.

If you think it could be useful, feel free to take a look at my fork at the localization branch.  I'm also happy to open a PR if you'd like to explore it further. 🙃

**Reply by johnmeshulam** [reply]
Hi, we actually initially vonsoderent both lingui and react-intl but opted for next-intl as our frontend projects use NextJS and the setup and overhead are extremely minimal due to how integrated it is.
